forum/models/base_model.py

```diff
@@ -3,40 +3,39 @@
 """
 
 from abc import ABC, abstractmethod
-from typing import Any, Dict, Optional
+from typing import Any, Optional
 
 from bson import ObjectId
-from pymongo.collection import Collection
+from pymongo.collection import Collection as PymongoCollection
 from pymongo.cursor import Cursor
 
-from forum.mongo import MongoBackend
+from forum.mongo import Database, get_database
+
+Collection = PymongoCollection[dict[str, Any]]
 
 
 class MongoBaseModel(ABC):
     """Abstract Class for Mongo model implementation"""
 
-    def __init__(
-        self,
-        collection_name: Optional[str] = None,
-        client: Optional[MongoBackend] = None,
-    ) -> None:
-        self.client: MongoBackend = client or MongoBackend(collection=collection_name)
+    MONGODB_DATABASE: Optional[Database] = None
+    COLLECTION_NAME: str = "default"
 
     @property
-    def _collection(self) -> Collection[Dict[str, Any]]:
-        """Get mongo db collection"""
-        return self.get_client.collection
+    def _collection(self) -> Collection:
+        return self.get_database()[self.COLLECTION_NAME]
 
-    @property
-    def get_client(self) -> MongoBackend:
-        """Get mongo client"""
-        return self.client
+    @classmethod
+    def get_database(cls) -> Database:
```

Comment:
can we make `get_database` class method private?

Reply:
Absolutely.